chia/rpc/wallet_rpc_api.py

```diff
@@ -1334,6 +1335,46 @@ async def verify_signature(self, request) -> EndpointResult:
         else:
             return {"isValid": False, "error": "Signature is invalid."}
 
+    async def check_ids(self, request) -> EndpointResult:
+        """
+        Check if a list of IDs is owned by the current wallet.
+        :param request: {"ids":["ID1", "ID2", ...], "type":"COIN/NFT/DID"}
+        :return: A list of ids owned by the current wallet
+        """
+        if "ids" not in request:
+            raise ValueError("Ids is required by check_ids.")
+        if "type" not in request:
+            raise ValueError("Type is required by check_ids.")
+        if request["type"].upper() == "COIN":
+            ids: List[bytes32] = [bytes32.fromhex(coin_id) for coin_id in request["ids"]]
+            coin_records: List[
+                WalletCoinRecord
+            ] = await self.service.wallet_state_manager.coin_store.get_multiple_coin_records(ids)
+            valid_ids = [r.coin.name().hex() for r in coin_records]
+        elif request["type"].upper() == "NFT":
+            ids = [
```

Comment:
@paninaro how many ids do you plan to check at a time?

Reply:
The primary use case I have in mind would be for checking if NFTs in an offer are owned by the user. We have a cap of ~10 NFTs in an offer when created from the GUI. So I would say ~20 would probably be the max.